diffpy/snmf/factorizers.py

```diff
@@ -0,0 +1,22 @@
+import numpy as np
+import scipy.optimize
+
+
+def lsqnonneg(coefficient, target):
+    """
+
+    Parameters
+    ----------
+    coefficient: ndarray
```

Comment:
I still think this is not quite right, right?  First, I think it is ok if you take a list as an input (since you arrayifiy it below). so the types could be array or list.

Second you are not doing any array shaping inside the code, so I think the function can only take 1d arrays, right?  What if someone passes in a 2D or 3D array, will the code crash?  You are certainly not testing for that.  So `ndarray` is not allowed in general.

Reply:
The scipy function requires that the coefficient array is 2 dimensional and the target array is one dimensional; the function raises an exception if this condition isn't met. If the input is a list it must be in the form of a list of lists so it becomes a 1 x N array instead of a 1 dimensional array.